pkg/apis/kf/v1alpha1/source_lifecycle.go

```diff
@@ -15,10 +15,81 @@
 package v1alpha1
 
 import (
+	"fmt"
+
+	"github.com/knative/build/pkg/apis/build/v1alpha1"
+	"github.com/knative/pkg/apis"
+	duckv1beta1 "github.com/knative/pkg/apis/duck/v1beta1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 // GetGroupVersionKind returns the GroupVersionKind.
 func (r *Source) GetGroupVersionKind() schema.GroupVersionKind {
 	return SchemeGroupVersion.WithKind("Source")
 }
+
+const (
+	// SourceConditionReady is set when the space is configured
+	// and is usable by developers.
+	SourceConditionReady = apis.ConditionReady
+	// SourceConditionBuildReady is set when the backing Build is ready.
+	SourceConditionBuildReady apis.ConditionType = "BuildReady"
+)
+
+func (status *SourceStatus) manage() apis.ConditionManager {
+	return apis.NewLivingConditionSet(
+		SourceConditionBuildReady,
+	).Manage(status)
+}
+
+// IsReady returns if the space is ready to be used.
+func (status *SourceStatus) IsReady() bool {
+	return status.manage().IsHappy()
+}
+
+// GetCondition returns the condition by name.
+func (status *SourceStatus) GetCondition(t apis.ConditionType) *apis.Condition {
+	return status.manage().GetCondition(t)
+}
+
+// InitializeConditions sets the initial values to the conditions.
+func (status *SourceStatus) InitializeConditions() {
+	status.manage().InitializeConditions()
+}
+
+// MarkBuildNotOwned marks the Build as not being owned by the Source.
+func (status *SourceStatus) MarkBuildNotOwned(name string) {
+	status.manage().MarkFalse(SourceConditionBuildReady, "NotOwned",
+		fmt.Sprintf("There is an existing Build %q that we do not own.", name))
+}
+
+// PropagateBuildStatus copies fields from the Build status to Space
+// and updates the readiness based on the current phase.
+func (status *SourceStatus) PropagateBuildStatus(build *v1alpha1.Build) {
+
+	if build == nil {
+		return
+	}
+
+	for _, condition := range build.Status.GetConditions() {
+
+		c := apis.Condition{
+			Type:               apis.ConditionType(string(condition.Type)),
+			Status:             condition.Status,
+			Severity:           apis.ConditionSeverity(string(condition.Severity)),
+			LastTransitionTime: condition.LastTransitionTime,
+			Reason:             condition.Reason,
+			Message:            condition.Message,
+		}
+
+		status.manage().SetCondition(c)
```

Comment:
I'm not sure it makes sense to copy the conditions of the build status up to this object--especially when we're supposed to pre-declare which conditions the manager manages

Reply:
Sounds good. I removed this